lib/src/google_fonts_base.dart

```diff
@@ -19,9 +19,10 @@ import '../google_fonts.dart';
 import 'google_fonts_descriptor.dart';
 import 'google_fonts_variant.dart';
 
-// Keep track of the loaded fonts in FontLoader for the life of the app
-// instance.
-final Set<String> _loadedFonts = {};
+// Keep track of the fonts that are attempted to be loaded fonts in FontLoader
+// for the life of the app instance. Once a font is attempted to load, it does
+// not need to be attempted to load again.
+final Set<String> _fontsAttemptedToLoad = {};
```

Comment:
Just a thought, maybe we should keep it as `_loadedFonts`, because if the HTTP request fails for whatever reason, we'd still want to make the request later. 

So maybe we can add it `familyWithVariantString` to the set immediately (like you're doing in this change) but then in the try catch, on error, we can _remove_ `familyWithVariantString` from the set.

We could even test this by mocking the http call to throw an error, calling `loadFontIfNecessary`, then mocking the http call to pass and calling it again and making sure that it loads the second time.